DataFormats/HcalRecHit/interface/HBHEChannelInfo.h

```diff
@@ -195,6 +206,11 @@ class HBHEChannelInfo
     // fC to GeV conversion factor (can depend on CAPID)
     double gain_[MAXSAMPLES];
 
+    // needed for the dark current
+    double darkCurrent_[MAXSAMPLES];
+    double fcByPE_[MAXSAMPLES];
+    double lambda_[MAXSAMPLES];
+
```

Comment:
These new members should not be arrays -- darkCurrent, fcByPE, and lambda do not change from one time slice to another. Make them scalars and use "setChannelInfo" function to set them instead of "setSample '